src/array_approx.rs

```diff
@@ -72,12 +74,51 @@ where
         if self.shape() != other.shape() {
             return false;
         }
+
         Zip::from(self)
             .and(other)
-            .all(|a, b| A::ulps_eq(a, b, epsilon.clone(), max_ulps))
+            .all(move |a, b| A::ulps_eq(a, b, epsilon.clone(), max_ulps))
     }
 }
 
+impl<A, S, D> ArrayBase<S, D>
+where
+    S: Data<Elem = A>,
+    D: Dimension,
+{
+    /// A test for equality that uses the elementwise absolute difference to compute the
+    /// approximate equality of two arrays.
+    ///
+    /// **Requires crate feature `"approx"`**
+    pub fn abs_diff_eq<B, S2>(&self, other: &ArrayBase<S2, D>, epsilon: A::Epsilon) -> bool
```

Comment:
IMO, this would be better with a single type parameter for simplicity:

```rust
pub fn abs_diff_eq<S2>(&self, other: &ArrayBase<S2, D>, epsilon: A::Epsilon) -> bool
where
    A: AbsDiffEq<S2::Elem>,
    A::Epsilon: Clone,
    S2: Data,
{}
```